src/otx/core/model/module/visual_prompting.py

```diff
@@ -243,22 +246,80 @@ def _inference_step(
+        """Configure metrics."""
         self.test_metric = MetricCollection(
             {
-                "IoU": BinaryJaccardIndex(),
-                "F1": BinaryF1Score(),
-                "Dice": Dice(),
-                "mAP": MeanAveragePrecision(iou_type="segm"),
+                "IoU": BinaryJaccardIndex().to(self.device),
+                "F1": BinaryF1Score().to(self.device),
+                "Dice": Dice().to(self.device),
+                "mAP": MeanAveragePrecision(iou_type="segm").to(self.device),
             },
         )
 
+    def on_train_start(self) -> None:
+        """Initialize reference infos before learn."""
+        self.model.initialize_reference_info()
+
+    def on_test_start(self) -> None:
+        """Load previously saved reference info."""
+        super().on_test_start()
+        if not self.model.load_latest_reference_info(self.device):
+            # TODO (sungchul): check fit_loop for OVModel # noqa: TD003
+            log.warning("No reference info found. `Learn` will be automatically excuted first.")
+            self.trainer.lightning_module.automatic_optimization = False
+            self.trainer.fit_loop.run()
+            # to use infer logic
+            self.training = False
+            self.model.training = False
+            # to set _combined_loader
+            self.trainer._evaluation_loop.setup_data()  # noqa: SLF001
+            self.trainer._evaluation_loop.reset()  # noqa: SLF001
+            self.model.load_latest_reference_info(self.device)
+
+    def on_predict_start(self) -> None:
+        """Load previously saved reference info."""
+        if not self.model.load_latest_reference_info(self.device):
+            # TODO (sungchul): check fit_loop for OVModel # noqa: TD003
+            log.warning("No reference info found. `Learn` will be automatically excuted first.")
+            self.trainer.lightning_module.automatic_optimization = False
+            self.trainer.fit_loop.run()
+            # to use infer logic
+            self.training = False
+            self.model.training = False
+            # to set _combined_loader
+            self.trainer._evaluation_loop.setup_data()  # noqa: SLF001
+            self.trainer._evaluation_loop.reset()  # noqa: SLF001
+            self.model.load_latest_reference_info(self.device)
+
     def on_train_epoch_start(self) -> None:
         """Skip on_train_epoch_start unused in zero-shot visual prompting."""
 
     def on_train_epoch_end(self) -> None:
         """Skip on_train_epoch_end unused in zero-shot visual prompting."""
+        if self.model.save_outputs:
+            reference_info = {
+                "reference_feats": self.model.reference_feats,
+                "used_indices": self.model.used_indices,
+            }
+            # save reference info
+            path_reference_info: Path = (
+                self.model.root_reference_info / time.strftime("%Y%m%d_%H%M%S") / "reference_info.pt"
```

Comment:
It might be good for making static variable for `reference_info.pt` and `pickle` extension in the future.

Reply:
You mean, override them every `learn`?